modules/ticket_selector/DisplayTicketSelector.php

```diff
@@ -374,24 +405,55 @@ protected function getTickets()
             && EE_Registry::instance()->CFG->template_settings->EED_Ticket_Selector->show_expired_tickets
         );
 
-        $ticket_query_args = array(
-            array('Datetime.EVT_ID' => $this->event->ID()),
-            'order_by' => array(
+        $ticket_query_args = [
+            [
+                'Datetime.EVT_ID' => $this->event->ID(),
+                'TKT_visibility' => ['<', EEM_Ticket::TICKET_VISIBILITY_NONE],
+            ],
+            'order_by' => [
                 'TKT_order'              => 'ASC',
                 'TKT_required'           => 'DESC',
                 'TKT_start_date'         => 'ASC',
                 'TKT_end_date'           => 'ASC',
                 'Datetime.DTT_EVT_start' => 'DESC',
-            ),
-        );
+            ],
+        ];
         if (! $show_expired_tickets) {
             // use the correct applicable time query depending on what version of core is being run.
             $current_time = method_exists('EEM_Datetime', 'current_time_for_query')
                 ? time()
                 : current_time('timestamp');
             $ticket_query_args[0]['TKT_end_date'] = array('>', $current_time);
         }
-        return EEM_Ticket::instance()->get_all($ticket_query_args);
+        /** @var EE_Ticket[] $tickets */
+        $tickets = EEM_Ticket::instance()->get_all($ticket_query_args);
+        // remove tickets based on their visibility and the current user's allowed access (crudely based on roles)
+        return array_filter($tickets, [$this, 'ticketVisibilityFilter']);
+    }
+
+
+    /**
+     * returns true if ticket visibility:
+     *  - is not NONE
+     *  - is less than MEMBERS_ONLY when user is NOT logged in
+     *  - is less than ADMINS_ONLY when user IS logged in but is NOT an admin
+     *  - is less than ADMIN_UI_ONLY when ticket selector is being viewed via a public UI
+     *
+     * @param EE_Ticket $ticket
+     * @return bool
+     * @throws EE_Error
+     * @throws ReflectionException
+     * @since   $VID:$
+     */
+    public function ticketVisibilityFilter(EE_Ticket $ticket): bool
+    {
+        $visibility = $ticket->visibility();
+        return $visibility < EEM_Ticket::TICKET_VISIBILITY_NONE
+            && ! (
+               ($visibility >= EEM_Ticket::TICKET_VISIBILITY_MEMBERS_ONLY && $this->current_user->ID === 0)
+               || ($visibility >= EEM_Ticket::TICKET_VISIBILITY_ADMINS_ONLY && ! $this->current_user_is_admin)
+               || ($visibility >= EEM_Ticket::TICKET_VISIBILITY_ADMIN_UI_ONLY && ! is_admin())
+           );
```

Comment:
This does not handle the case for `TICKET_VISIBILITY_PUBLIC` because it's the default value, right?

Reply:
that's correct... 
although your comment got me thinking about how I'm using the constants as thresholds here
the levels are basically:

- 100 PUBLIC
- 200 MEMBERS_ONLY
- 300 ADMINS_ONLY
- 400 ADMIN_UI_ONLY
- 500 NONE

but the way I have the thresholds set in that filter function looks like PUBLIC is everything up to 199, and similarly MEMBERS_ONLY goes up to 299, etc...

should it be the other way around though?
ie should those values be the upper limit for each threshold:

- 0 - 100 PUBLIC
- 101 - 200 MEMBERS_ONLY
- 201 - 300 ADMINS_ONLY
- 301 - 400 ADMIN_UI_ONLY
- 401 - 500 NONE

or the lower level:

-   0 -  99 ???
- 100 - 199 PUBLIC
- 200 - 299 MEMBERS_ONLY
- 300 - 399 ADMINS_ONLY
- 400 - 499 ADMIN_UI_ONLY
- 500+      NONE

what are your thoughts on this?